src/router/definitions.ts

```diff
@@ -0,0 +1,54 @@
+export type Route =
+  | { type: "404" }
+  | {
+      type: "faucet";
+      params: {
+        activeView:
+          | { type: "form" }
+          | { type: "withdraw"; params: { amount: string | null } };
+      };
+    }
+  | { type: "home" }
+  | { type: "boot" }
+  | { type: "profile"; params: { addressOrName: string } }
+  | { type: "projects"; params: ProjectsParams }
+  | {
+      type: "registration";
+      params: {
+        activeView:
+          | {
+              type: "validateName";
+            }
+          | {
+              type: "checkNameAvailability";
+              params: { nameOrDomain: string; owner: string | null };
+            }
+          | {
+              type: "register";
+              params: { nameOrDomain: string; owner: string | null };
+            }
+          | {
+              type: "view";
+              params: { nameOrDomain: string; retry: boolean };
+            };
+      };
+    }
+  | { type: "seeds"; params: { host: string } }
+  | { type: "vesting" };
+
+export interface ProjectsParams {
+  content?: string;
+  hash?: string | null;
+  issue?: string | null;
+  patch?: string | null;
+  path?: string | null;
+  peer?: string | null;
+  profile?: string | null;
+  revision?: string | null;
+  route?: string | null;
+  search?: string | null;
+  seed?: string | null;
+  urn: string;
+}
```

Comment:
Why are we allowing both `null` and `undefined` in all of these?

Reply:
I ended up with both `null` and `undefined`, so we would be able to define some props as `null` in the `pathToRoute` fn but at the same time we could only pass the params we wanted to define, instead of declaring every route param.
Similar to what we had before with the `BrowseTo` interface https://github.com/radicle-dev/radicle-interface/blob/87c1d0f6e2a06b2f6572a1fdcd546a1ba2d47aea/src/project.ts#L121

Reply:
fixed by dfe546f49cda95e1dde641a87715950245932de3